### ISSUE TITLE
Enhanced behaviour of quickfix window

### ISSUE THREAD

**Issue by RishabhRD** [neovim/neovim#12906]
- neovim version: NVIM v0.5.0-691-g542022aae
- Operating System: Arch linux
- Terminal name/version: st
- `$TERM`: st-256color

### Actual behaviour

Functions like: vim.lsp.buf.references() opens quickfix window. However, that quickfix window doesn't get focused automatically.
Also, when we choose any item from quickfix window by pressing enter, we only jump to desired location. However, quickfix window still doesn't close.

### Expected behaviour

When functions like vim.lsp.buf.references() opens a quickfix window, the quickfix window should get focused automatically. And after choosing an item from quickfix window should close quickfix window.

Not closing quickfix window automatically is like overhead for users to close window by themselves.

**Comment by erw7** [comment]
I don't know why it was changed in this way, but #11452 seems to have changed it so that `<c-w>p` is used to move to the previous window after the quickfix window is opened. This may be a behavior that needs to be fixed, as `vimgrep` and others make it stay in the quickfix window.

cc: @norcalli

https://github.com/neovim/neovim/blob/569e75799d7015b15631c80cee1feec561f29df7/runtime/lua/vim/lsp/callbacks.lua#L115-L120

I think you can close the quickfix window after selecting an item with the following settings. You can change the key to map from `<tab>` if you like.

```vim
autocmd FileType qf if getwininfo(win_getid())[0].loclist | execute 'nnoremap <silent><buffer><tab> :.ll <bar> lclose<cr>' | end
autocmd FileType qf if !getwininfo(win_getid())[0].loclist | execute 'nnoremap <silent><buffer><tab> :.cc <bar> cclose<cr>' | end
```

**Comment by OliverNChalk** [comment]
This issue is still present in the latest release, did you manage to successfully replicate your desired behaviour @RishabhRD ?

**Comment by mjlbach** [comment]
* the quickfix window should get focused automatically. 
This was closed in #14878

* And after choosing an item from quickfix window should close quickfix window.
We opted not to do this, as many users want to toggle through the list of references/jump locations to find the correct one, and there's no distinction between toggling through and "choosing" an entry.
